src/mame/taito/taito_f3.h

```diff
@@ -167,136 +172,220 @@ class taito_f3_state : public driver_device
+		inline u8 blend_mask() const { return BIT(mix_value, 14, 2); };
+		inline bool blend_a() const { return mix_value & 0x4000; };
+		inline bool blend_b() const { return mix_value & 0x8000; };
+
+		inline bool operator<(const mixable& rhs) const noexcept { return this->prio() < rhs.prio(); };
+		inline bool operator>(const mixable& rhs) const noexcept { return this->prio() > rhs.prio(); };
+
+		virtual inline u16 palette_adjust(u16 pal) const { return pal; };
+		virtual inline int y_index(int y) const;
+		virtual inline int x_index(int x) const;
+		virtual bool blend_select(const u8 *line_flags, int x) const { return false; };
+
+		virtual bool used(int y) const { return true; };
+		u8 debug_index{0};
+		virtual const char* debug_name() { return "MX"; };
+	};
+
+	struct sprite_inf : mixable {
+		// alpha mode in 6000
+		// mosaic enable in 6400
+		// line enable, clip settings in 7400
+		// priority in 7600
+		bool blend_select_v{false}; // 7400 0xf000
+		bool blend_select(const u8 *line_flags, int x) const override { return blend_select_v; };
+
+		u8 (*sprite_pri_usage)[256]{nullptr};
+		bool used(int y) const override { return (*sprite_pri_usage)[y] & (1<<debug_index); }
+		const char* debug_name() override { return "SP"; };
+	};
+
+	struct pivot_inf : mixable {
+		u8 pivot_control{0};     // 6000
+		bool blend_select_v{false};
+		bool blend_select(const u8 *line_flags, int x) const override { return blend_select_v; };
+		// mosaic enable in 6400
+		u16 pivot_enable{0}; // 7000 - what is in this word ?
+		// mix info from 7200
+		bool use_pix() const { return pivot_control & 0xa0; };
+
+		u16 reg_sx{0};
+		u16 reg_sy{0};
+		inline int y_index(int y) const override;
+		inline int x_index(int x) const override;
+		const char* debug_name() override { return "PV"; };
+	};
+
+	struct playfield_inf : mixable {
+		u16 colscroll{0};            // 4000
+		bool alt_tilemap{false};     // 4000
+		// mosaic enable in 6400
+		fixed8 x_scale{0x80};        // 8000
+		fixed8 y_scale{0};           // 8000
+		u16 pal_add{0};              // 9000
+		fixed8 rowscroll{0};         // a000
+
+		fixed8 reg_sx{0};
+		fixed8 reg_sy{0};
+		fixed8 reg_fx_y{0};
+		fixed8 reg_fx_x{0};
+
+		u16 width_mask{0};
+
+		inline u16 palette_adjust(u16 pal) const override;
+		inline int y_index(int y) const override;
+		inline int x_index(int x) const override;
+		bool blend_select(const u8 *line_flags, int x) const override { return BIT(line_flags[x], 0); };
+		const char* debug_name() override { return "PF"; };
+	};
+
+	struct pri_mode {
+		u8 src_prio{0};
+		u8 dst_prio{0};
+		u8 src_blendmode{0xff};
+		u8 dst_blendmode{0xff};
+	};
+
+	struct f3_line_inf {
+		int y{0};
+		int screen_y{0};
+		pri_mode pri_alp[432]{};
+		// 5000/4000
+		clip_plane_inf clip[NUM_CLIPPLANES];
+		// 6000 - pivot_control, sprite alpha
+		u16 maybe_sync_reg{0};
+		bool no_opaque_dest{false};
+		// 6200
+		u8 blend[4]{0}; // less 0 - 8 more
```

Comment:
FYI initializing arrays like this, it will only initialize the 1st index, and fill the rest with 0. Best use {} like you did elsewhere.

see:
int m_array[4] { 1 };
for (int i = 0; i < 4 ; i++) printf("%d ",m_array[i]);
-> 1 0 0 0

Reply:
thanks; fixed.